src/app/Main/Body/QueryPage/Results/Content-Results.component.tsx

```diff
@@ -1,126 +1,125 @@
+                ? data.uniqueValues[colAttribute]
+                : filter.selectedFilter[colAttribute])
+
+            const isolateColValues: string[] = [t("Results.TableHead")];
+            const isolateRowValues: string[] = [t("Results.TableHead")];
+            isolateCountProp.groups.forEach((isolateCount) => {
+                isolateColValues.push(isolateCount[colAttribute]);
+                isolateRowValues.push(isolateCount[rowAttribute]);
             });
-            setColumnAttributes(colValues);
-            setTable({
-                ...table,
-                statisticDataAbsolute: rowsWithIsolates,
-                statisticDataPercent: rowsWithPercent,
+            const uniqIsolateColValues = _.uniq(isolateColValues);
+            const uniqIsolateRowValues = _.uniq(isolateRowValues);
+
+            const statisticTableDataAbs: Record<string, string>[] = [];
+
+            rowValues.forEach((rowElement) => {
+                const tempStatTable: Record<string, string> = {};
+                tempStatTable.name = rowElement;
+                colValues.forEach((colElement) => {
+                    if (
+                        _.includes(uniqIsolateColValues, colElement) &&
+                        _.includes(uniqIsolateRowValues, rowElement)
+                    ) {
+                        isolateCountProp.groups.forEach((isolateGroup) => {
+                            if (
+                                (isolateGroup[rowAttribute] === rowElement ||
+                                    isolateGroup[rowAttribute] === undefined) &&
+                                (isolateGroup[colAttribute] === colElement ||
+                                    isolateGroup[colAttribute] === undefined)
+                            ) {
+                                const statTableKey =
+                                    isolateGroup[colAttribute] === undefined
+                                        ? t("Results.TableHead")
+                                        : isolateGroup[colAttribute];
+                                tempStatTable[
+                                    statTableKey
+                                ] = (isolateGroup.count as unknown) as string;
+                            }
+                        });
+                    } else {
+                        tempStatTable[colElement] = "0";
+                    }
+                });
+                statisticTableDataAbs.push(tempStatTable);
             });
-        } else if (colAttribute.length === 0 && rowAttribute.length !== 0) {
-            const rowsWithIsolates = generateRowsWithIsolates(
-                dataToCountIsolates,
-                colAttribute,
-                rowAttribute,
-                rowValues,
-                [t("Results.TableHead")],
-                "row"
+
+            const nrOfSelectedIsolates = isolateCountProp.totalNumberOfIsolates;
+            const rowsWithPercent = calculateRowsWithPercent(

```

Comment:
consider rename to statisticTableDataRel or something

Reply:
Done. Commit "Rename rowsWithPercent"